pkg/gatewayserver/io/mqtt/format_protobufv2.go

```diff
@@ -66,6 +66,11 @@ type protobufv2 struct {
 	topics.Layout
 }
 
+// TimePtr converts time to a pointer.
+func TimePtr(x time.Time) *time.Time {
+	return &x
```

Comment:
Does this need `UTC` ? Maybe that's how we end up with these weirdly deformed timestamps of gateway times (not saying this call site necessarily, but _maybe_...)

Reply:
Good question. The reason I added UTC in the other test was that when it was dereferenced by the assertions package, it takes local time. Let me check what the GS does in general with time and get back to you.

Reply:
I'd say this is out of scope for this PR, but indeed good to check.